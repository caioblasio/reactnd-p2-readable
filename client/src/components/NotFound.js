import React, { Component } from 'react';

import { withRouter } from 'react-router-dom';

import { withStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardContent from '@material-ui/core/CardContent';


import Typography from '@material-ui/core/Typography';

import Divider from '@material-ui/core/Divider';
import CardActions from '@material-ui/core/CardActions';
import Button from '@material-ui/core/Button';

const styles = theme => ({
  card: {
    width: '70%',
    margin: '0 auto',
  },
  link: {
    ...theme.link,
    outline: 0,
  },
  actions: {
    display: 'flex',
    alignItems: 'center',
  },
});

class NotFound extends Component {

  render() {

    const { classes, history } = this.props;
    return (
      <Card className={classes.card}>
          <CardHeader
            title="Page Not Found :("
          />
          <CardContent>
            <Typography component="p">
                Maybe the page you are looking for has been removed, or you typed in the wrong URL
            </Typography>
          </CardContent>
          <CardActions className={classes.actions} disableActionSpacing>
            <Button size="small" color="primary" onClick={() => {history.goBack()}}>
              Go Back
            </Button>
            <Button size="small" color="primary" onClick={() => {history.push('/')}}>
              Go To Homepage
            </Button>
          </CardActions>
      </Card>
    )
  }
}

export default withStyles(styles)(withRouter(NotFound))