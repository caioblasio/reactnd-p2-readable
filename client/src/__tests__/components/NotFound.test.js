import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import NotFound from '../../components/NotFound';
import { MemoryRouter, Router } from 'react-router-dom';
import Button from '@material-ui/core/Button';


describe('<NotFound />', () => {

    it('renders the component', () => {
      const wrapper = shallow(
      <MemoryRouter>
        <NotFound />
      </MemoryRouter>
      );
      const component = wrapper.dive();

      expect(toJson(component.dive())).toMatchSnapshot();
    });

    it('Click on Go To Home button', () => {

      const mounted = mount(
        <MemoryRouter initialEntries={['/notexist']}>
          <NotFound />
        </MemoryRouter>
      );

      mounted.find(Button).last().simulate('click');
      expect(mounted.find(Router).props('history').history.location.pathname).toBe('/')
      expect(mounted.find(Router).props('history').history.length).toBe(2);
    });

    it('Click on Back button', () => {

      const mounted = mount(
        <MemoryRouter initialEntries={['/', '/notexist']} initialIndex={1}>
          <NotFound />
        </MemoryRouter>
      );

      mounted.find(Button).first().simulate('click');
      expect(mounted.find(Router).props('history').history.location.pathname).toBe('/');

    });

});