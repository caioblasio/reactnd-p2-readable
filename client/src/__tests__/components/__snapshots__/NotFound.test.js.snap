// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> renders the component 1`] = `
<withRouter(NotFound)
  classes={
    Object {
      "actions": "withRouter-NotFound--actions-3",
      "card": "withRouter-NotFound--card-1",
      "link": "withRouter-NotFound--link-2",
    }
  }
/>
`;
